src/api/Item.js

```diff
@@ -302,37 +329,68 @@ class Item extends Base {
             return Promise.reject();
         }
 
-        this.errorCode = ERROR_CODE_SHARE_ITEM;
-        const { id, permissions }: BoxItem = item;
-        if (!id || !permissions) {
-            errorCallback(getBadItemError(), this.errorCode);
-            return Promise.reject();
-        }
+        try {
+            const { id, permissions }: BoxItem = item;
+            this.id = id;
+            this.successCallback = successCallback;
+            this.errorCallback = errorCallback;
 
-        const { can_share, can_set_share_access }: BoxItemPermission = permissions;
-        if (!can_share || !can_set_share_access) {
-            errorCallback(getBadPermissionsError(), this.errorCode);
-            return Promise.reject();
-        }
+            this.validateRequest(id, permissions);
 
-        const cache: APICache = this.getCache();
```

Comment:
Removing the cache from `share()` and `updateSharedLink()` because we should always make an API call for creating, removing, and updating shared links, even if the item is already cached. (Also, this breaks the sharing modals in ContentExplorer and ContentPicker.)